src/bundle/Resources/public/js/scripts/core/dropdown.js

```diff
@@ -151,52 +143,82 @@
         }
 
         getItemsContainerHeight(isItemsContainerAbove) {
-            const DROPDOWN_MARGIN = 16;
-            const SELECTOR_MODAL = '.modal[aria-modal=true]';
+            const DROPDOWN_MARGIN = 32;
             const documentElementHeight = global.innerHeight;
-            const itemsContainerTop = this.itemsContainer.getBoundingClientRect().top;
+            const { top, bottom } = this.selectedItemsContainer.getBoundingClientRect();
 
             if (isItemsContainerAbove) {
-                return this.container.querySelector(SELECTOR_SELECTION_INFO).getBoundingClientRect().top - DROPDOWN_MARGIN;
+                return top - DROPDOWN_MARGIN;
             }
 
-            if (this.itemsContainer.closest(SELECTOR_MODAL)) {
-                return itemsContainerTop - DROPDOWN_MARGIN;
+            return documentElementHeight - bottom - DROPDOWN_MARGIN;
+        }
+
+        removeExpandingClass(event) {
+            if (event.propertyName === 'transform') {
+                this.container.classList.remove('ibexa-dropdown--is-expanding');
+                this.itemsContainer.removeEventListener('transitionend', this.removeExpandingClass, false);
             }
+        }
+
+        collapseDropdown() {
+            this.itemsContainer.addEventListener('transitionend', this.removeExpandingClass, false);
 
-            return documentElementHeight - itemsContainerTop - DROPDOWN_MARGIN;
+            setTimeout(() => {
+                this.container.classList.remove('ibexa-dropdown--is-expanded');
+                this.hideOptions();
+                this.fireValueChangedEvent();
+            }, 1);
+        }
+
+        expandDropdown() {
+            const viewportHeight = window.innerHeight || document.documentElement.clientHeight;
+            const { left, top, bottom } = this.selectedItemsContainer.getBoundingClientRect();
+            const isItemsContainerAbove = bottom + ITEMS_LIST_MIN_HEIGHT > viewportHeight;
+
+            this.itemsFixedWrapperContainer.style.left = `${left}px`;
+            this.itemsFixedWrapperContainer.style.top = `${isItemsContainerAbove ? top : bottom}px`;
+
+            this.itemsFixedWrapperContainer.classList
+                .toggle('ibexa-dropdown__items-fixed-wrapper--position-top', isItemsContainerAbove);
+
+            this.itemsContainer.style['max-height'] = `${this.getItemsContainerHeight(isItemsContainerAbove)}px`;
+
+            this.container.classList.add('ibexa-dropdown--is-expanding');
+            this.itemsFilterInput
+                .closest('.ibexa-input-text-wrapper')
+                .querySelector('.ibexa-input-text-wrapper__action-btn--clear')
+                .click();
+            this.itemsFilterInput.focus();
+
+            setTimeout(() => {
+                this.container.classList.add('ibexa-dropdown--is-expanded');
+            }, 1);
```

Comment:
Is this solution robust? Do we need `setTimeout`?

Reply:
Yes, as we need first to add class that will change display to block, we can't run both changing display to block and changing scale to 1 in same frame, as in such case transition won't be visible